src/main/java/net/mcreator/element/types/Feature.java

```diff
@@ -0,0 +1,87 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types;
+
+import net.mcreator.blockly.data.BlocklyLoader;
+import net.mcreator.blockly.feature.BlocklyToFeature;
+import net.mcreator.element.BaseType;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.BiomeEntry;
+import net.mcreator.element.parts.Procedure;
+import net.mcreator.element.types.interfaces.ICommonType;
+import net.mcreator.generator.blockly.BlocklyBlockCodeGenerator;
+import net.mcreator.generator.blockly.OutputBlockCodeGenerator;
+import net.mcreator.generator.blockly.ProceduralBlockCodeGenerator;
+import net.mcreator.generator.mapping.NameMapper;
+import net.mcreator.generator.template.IAdditionalTemplateDataProvider;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@SuppressWarnings("unused") public class Feature extends GeneratableElement implements ICommonType {
+
+	public String generationStep;
+	public List<String> restrictionDimensions;
+	public List<BiomeEntry> restrictionBiomes;
+	public Procedure generateCondition;
+	public String featurexml;
+
+	public Feature(ModElement element) {
+		super(element);
+
+		this.generationStep = "SURFACE_STRUCTURES";
+		this.restrictionDimensions = new ArrayList<>();
+		this.restrictionBiomes = new ArrayList<>();
+	}
+
+	@Override public @Nullable IAdditionalTemplateDataProvider getAdditionalTemplateData() {
+		return additionalData -> {
+			var blocklyBlockCodeGenerator = new BlocklyBlockCodeGenerator(
+					BlocklyLoader.INSTANCE.getFeatureBlockLoader().getDefinedBlocks(),
+					this.getModElement().getGenerator().getTemplateGeneratorFromName("features"), additionalData);
+
+			var blocklyToFeature = new BlocklyToFeature(this.getModElement().getWorkspace(), this.getModElement(),
+					this.featurexml, this.getModElement().getGenerator().getTemplateGeneratorFromName("features"),
+					new ProceduralBlockCodeGenerator(blocklyBlockCodeGenerator),
+					new OutputBlockCodeGenerator(blocklyBlockCodeGenerator));
+
+			additionalData.put("placementcode", blocklyToFeature.getGeneratedCode());
+			additionalData.put("configurationcode", blocklyToFeature.getFeatureConfigurationCode());
+
+			String featureType = blocklyToFeature.getFeatureType();
+			var featureMapper = new NameMapper(this.getModElement().getWorkspace(), "features");

```

Comment:
I would use g.map from GeneratorWrapper directly in the template, as this is more of a decision of the generator author if they want to use mapping for this, or some if conditions directly I would say